common/deftype/defvalue.cpp

```diff
@@ -435,8 +435,10 @@ void MemoryValue::pushDecimalValue()
 
 void MemoryValue::serialize(MemoryBuffer &tgt)
 {
-    tgt.append(val.length());
-    tgt.append(val.length(), val.get());
+    size32_t serialLen = (size32_t)val.length();
```

Comment:
It is for backward compatibility - and also, at the moment we don't support values > 2^32.
Having said that, I think this is only called in one place, and I'm not sure it is ever really used.